### PR TITLE
feat: remove `Value` from the public API, make it internal for plugins

### PR DESCRIPTION
**Pull request by wyfo**
@Mallets @milyin

### CODE REVIEW

File: zenoh/src/api/query.rs [hunk 1]
```diff
@@ -79,27 +79,63 @@ impl Default for QueryConsolidation {
...
+impl ReplyError {
+    /// Gets the payload of this ReplyError.
+    #[inline]
+    pub fn payload(&self) -> &ZBytes {
+        &self.payload
+    }
+
+    /// Gets the payload of this ReplyError.
+    #[inline]
+    pub fn payload_mut(&mut self) -> &mut ZBytes {
```
Comment (Mallets) [inline zenoh/src/api/query.rs:98]: Is `payload_mut` really necessary?

File: zenoh/src/api/builders/sample.rs [hunk 2]
```diff
@@ -99,6 +93,14 @@ impl SampleBuilder<SampleBuilderPut> {
             _t: PhantomData::<SampleBuilderPut>,
         }
     }
+
+    pub fn payload<IntoZBytes>(mut self, payload: IntoZBytes) -> Self
```
Comment (Mallets) [inline zenoh/src/api/builders/sample.rs:97]: I don't think this is needed. Actual payload is already provided in the `put()` function.

File: zenoh/src/api/query.rs (hunk 1 above)
Comment (wyfo) [reply zenoh/src/api/query.rs:98]: It was copied from `Value`, but I think we can remove it yes, so I will do it.

File: zenoh/src/api/builders/sample.rs (hunk 2 above)
Comment (Mallets) [reply zenoh/src/api/builders/sample.rs:97]: You are right. In that case it makes sense to keep it.

**Comment by Mallets** [comment]
As it stands this PR LGTM. However CI should be green before approving and merging (modulo the memory leak test that still needs to be fixed).

**Review by Mallets — APPROVED** [review]

**Pull request merged**